bpf/bpf_network.c

```diff
@@ -19,8 +19,7 @@ int from_network(struct __ctx_buff *ctx)
 
 	if ((ctx->mark & MARK_MAGIC_HOST_MASK) == MARK_MAGIC_DECRYPT) {
 		send_trace_notify(ctx, TRACE_FROM_NETWORK, get_identity(ctx), 0, 0,
-				  ctx->ingress_ifindex,
-				  TRACE_REASON_ENCRYPTED, TRACE_PAYLOAD_LEN);
+				  ctx->ingress_ifindex, 0, TRACE_PAYLOAD_LEN);
```

Comment:
Do we need to instead emit the `TRACE_REASON_ENCRYPTED` trace below, for when packets come in without a mark? We may need to parse the packet to check that it's actually an ESP packet as we may also receive unencrypted packets.

Reply:
Yeah, I was wondering if it is worth it to make sure the packet is encrypted or not in here by parsing the protocol number, but we anyways do that with `do_decrypt` here. I'll try to refactor do_decrypt to return protocol number then.

Reply:
Fixed in 9ca7d1170dc5eb96a06fd4f8aad0e402cfe9d52b